From 1eb50c745df2b5a3d703add491c2a716584c3c6f Mon Sep 17 00:00:00 2001
From: Luc Michel <luc.michel@amd.com>
Date: Wed, 30 Sep 2026 11:07:53 +0200
Subject: [DTS PATCH] Yocto boards with alternative UART order

This commit creates the alternative boards for Yocto with a change in
the UART instantiation order on the QEMU command line.

Signed-off-by: Luc Michel <luc.michel@amd.com>

Upstream-Status: Inappropriate [OE-Specific]

Signed-off-by: Mark Hatle <mark.hatle@amd.com>
---
 alternative-uart-order.dtsi          | 16 ++++++++++++++++
 board-versal-net-psx-spp-1.4-alt.dts |  2 ++
 board-versal-net-psx-virt-alt.dts    |  3 +++
 board-versal-pmc-virt-alt.dts        |  6 ++++++
 board-versal-pmx-virt-alt.dts        |  6 ++++++
 board-versal-ps-vck190-alt.dts       |  2 ++
 board-versal-ps-virt-alt.dts         |  2 ++
 7 files changed, 37 insertions(+)
 create mode 100644 alternative-uart-order.dtsi
 create mode 100644 board-versal-net-psx-spp-1.4-alt.dts
 create mode 100644 board-versal-net-psx-virt-alt.dts
 create mode 100644 board-versal-pmc-virt-alt.dts
 create mode 100644 board-versal-pmx-virt-alt.dts
 create mode 100644 board-versal-ps-vck190-alt.dts
 create mode 100644 board-versal-ps-virt-alt.dts

diff --git a/alternative-uart-order.dtsi b/alternative-uart-order.dtsi
new file mode 100644
index 000000000..4fe8fa6d1
--- /dev/null
+++ b/alternative-uart-order.dtsi
@@ -0,0 +1,16 @@
+&ppu0_mdm_uart {
+    chardev = "serial2";
+};
+
+&ppu1_mdm_uart {
+    chardev = "serial3";
+};
+
+&serial0 {
+    chardev = "serial0";
+};
+
+&serial1 {
+    chardev = "serial1";
+};
+
diff --git a/board-versal-net-psx-spp-1.4-alt.dts b/board-versal-net-psx-spp-1.4-alt.dts
new file mode 100644
index 000000000..0818ebed0
--- /dev/null
+++ b/board-versal-net-psx-spp-1.4-alt.dts
@@ -0,0 +1,2 @@
+#include "board-versal-net-psx-spp-1.4.dts"
+#include "alternative-uart-order.dtsi"
diff --git a/board-versal-net-psx-virt-alt.dts b/board-versal-net-psx-virt-alt.dts
new file mode 100644
index 000000000..90ae32b18
--- /dev/null
+++ b/board-versal-net-psx-virt-alt.dts
@@ -0,0 +1,3 @@
+#include "board-versal-net-psx-virt.dts"
+#include "alternative-uart-order.dtsi"
+
diff --git a/board-versal-pmc-virt-alt.dts b/board-versal-pmc-virt-alt.dts
new file mode 100644
index 000000000..bda73a858
--- /dev/null
+++ b/board-versal-pmc-virt-alt.dts
@@ -0,0 +1,6 @@
+#include "board-versal-pmc-virt.dts"
+
+#ifndef MULTI_ARCH
+#include "alternative-uart-order.dtsi"
+#endif
+
diff --git a/board-versal-pmx-virt-alt.dts b/board-versal-pmx-virt-alt.dts
new file mode 100644
index 000000000..f2cdba6db
--- /dev/null
+++ b/board-versal-pmx-virt-alt.dts
@@ -0,0 +1,6 @@
+#include "board-versal-pmx-virt.dts"
+
+#ifndef MULTI_ARCH
+#include "alternative-uart-order.dtsi"
+#endif
+
diff --git a/board-versal-ps-vck190-alt.dts b/board-versal-ps-vck190-alt.dts
new file mode 100644
index 000000000..09b06fd9e
--- /dev/null
+++ b/board-versal-ps-vck190-alt.dts
@@ -0,0 +1,2 @@
+#include "board-versal-ps-vck190.dts"
+#include "alternative-uart-order.dtsi"
diff --git a/board-versal-ps-virt-alt.dts b/board-versal-ps-virt-alt.dts
new file mode 100644
index 000000000..35284ae9d
--- /dev/null
+++ b/board-versal-ps-virt-alt.dts
@@ -0,0 +1,2 @@
+#include "board-versal-ps-virt.dts"
+#include "alternative-uart-order.dtsi"
-- 
2.55.0

